### ISSUE TITLE
Provider for coordinates over land

### ISSUE THREAD

**Issue by shacker** [joke2k/faker#857]
I am willing to make a pull request, but want to get your input first.

Problem: Earth is mostly covered with water. Therefore, choosing coordinates at random results in the majority of them not being on land. Maps produced from locations faked with the current coordinate providers in faker are not very useful.

Proposal: Create a new provider that instead pulls from a list of lat/long coords known to be over land.

I have this working locally, using the first 500 cities from the list [here](http://gael-varoquaux.info/misc/gps-coordinates-for-the-worlds-major-cities.html).

Questions:

- Are you interested in including this new provider in faker?

- Since this has to be a data set, and is not language-specific, where is the best place to store a long list of tuples in the repo? Should I just include them directly in a new method in `faker/providers/__init__.py` ? Or reference a supporting data file elsewhere in the repo (and if so, where?)

- How many locations should it include? 500 OK? Or more like 100?

- Can I use the data from the link above, which was scraped by that person's python script? Or should I hand-pick the locations myself? Or use some other more open-source set?

**Comment by shacker** [comment]
Also: most web mapping tools expect coordinates in decimal format, whereas most lists of global coordinates (wikipedia, infoplease) are in degree format, making them not directly useful for this project without a lot of lookup/conversion work.

**Comment by fcurella** [comment]
Hi @shacker,

I think we should cover the most common case scenario, which I think is location on land. But I also want to leave the option option for applications that are expected to handle locations at see. So we could have two lists, or one list + the old behavior, and switch according to an argument.

To answer your questions:

> Are you interested in including this new provider in faker?

Yes

> Since this has to be a data set, and is not language-specific, where is the best place to store a long list of tuples in the repo

We already have the functionality in the address provider, but we can move it to an external `geo` provider, as long as the current methods (`geo_coordinate`, `latitude`, `latitude`) maintain their name so users' code doesn't break.

> Should I just include them directly in a new method in faker/providers/__init__.py ?

Even if it's not localized, you'd stiil have to create `faker/geo/__init__.py` and at least `faker/geo/en_US/__init__.py`

> Or reference a supporting data file elsewhere in the repo (and if so, where?)

That's a feature I wanted to implement for all providers, but never had time to. See https://github.com/joke2k/faker/issues/178. If you can come up with a good solution that's generic enough for all providers, please submit a PR! :)

> How many locations should it include? 500 OK? Or more like 100?

The more the better.

> Can I use the data from the link above, which was scraped by that person's python script? Or should I hand-pick the locations myself? Or use some other more open-source set?

My understanding is that that file is the result of scraping a website with 'all right reserved', and therefore we do won't have the permission to use that data. When we're in doubt about permissions, the answer is no. We'd need either an open-source set, or something scraped from a page with an explicit permissive license such as Public Domain or Creative Commons by Attribution.

**Comment by shacker** [comment]
Great! A few notes below, then I'll get started. 

I think making a new `geo` provider makes sense. No intention of breaking the existing functionality - just adding a new `location_on_land` provider. I'll move the existing ones over while keeping their names (fortunately faker calls don't namespace the provider collections so nothing will break). 

Fully understood about your caution wrt open source data sets. I'll keep looking, and will select them all by hand if necessary.

**Comment by shacker** [comment]
OK, I've located a good open source database at [geonames](http://www.geonames.org/export/). The [readme](http://download.geonames.org/export/dump/readme.txt) says it's Creative Commons Attribution 3.0 License, which in turn says:

```
You are free to:
Share — copy and redistribute the material in any medium or format
Adapt — remix, transform, and build upon the material
for any purpose, even commercially.
```

I've parsed and processed one of their city lists to generate a tuple of `lat, long, placename, countrycode, timezone)`. I'm thinking that with no args, it will return a tuple of `(lat, long)`. The additional columns will be included if args for each are present. 

That's what I'm going on so far, unless you have further input.
